packages/common/src/api/tan-query/useTopArtistsInGenre.ts

```diff
@@ -0,0 +1,45 @@
+import { useQuery, useQueryClient } from '@tanstack/react-query'
+import { useDispatch } from 'react-redux'
+
+import { userMetadataFromSDK } from '~/adapters'
+import { transformAndCleanList } from '~/adapters/utils'
+import { useAudiusQueryContext } from '~/audius-query'
+
+import { QUERY_KEYS } from './queryKeys'
+import { Config } from './types'
+import { primeUserData } from './utils/primeUserData'
+
+const ARTISTS_PER_GENRE_LIMIT = 15
+
+type UseTopArtistsInGenreArgs = {
+  genre: string
+  limit?: number
+  offset?: number
+}
+
+export const useTopArtistsInGenre = (
+  args: UseTopArtistsInGenreArgs,
+  config?: Config
+) => {
+  const { audiusSdk } = useAudiusQueryContext()
+  const queryClient = useQueryClient()
+  const dispatch = useDispatch()
+  const { genre, limit = ARTISTS_PER_GENRE_LIMIT, offset } = args
+
+  return useQuery({
+    queryKey: [QUERY_KEYS.topArtistsInGenre, genre, limit, offset],
```

Comment:
if we have limit and offset, should we just be using `useInfiniteQuery`? or maybe not because the lineup or paginated usage isn't expecting it in that format?

Reply:
agreed, migrating this to useInfinite